### PR TITLE
Add more content to site and redesign it.

### PR DESCRIPTION
**Pull request by brendandahl**
Needs some follow up:
- fix all the recipes so they look consistent!
- break out templates and recipe functions into there own files
- minify js/css
- use sass instead of css?

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -30,25 +127,69 @@ gulp.task('lint', function lint() {
...
-    .pipe(plugins.swig({
-      data: {
-        package: require('./package.json'),
-        recipes: recipes,
-        ghBase: 'https://github.com/digitarald/serviceworker-cookbook/blob/master/'
-      }
-    }))
-    .pipe(gulp.dest('./dist'));
+  var intros = [];
+  recipeSlugs.forEach(function(recipe) {
```
Comment (digitarald) [inline gulpfile.js]: Any reason not to use `Array::map`?

File: gulpfile.js [hunk 2]
```diff
@@ -1,23 +1,120 @@
...
-var srcRecipes = recipes.map(function makePath(name) {
+var srcRecipes = recipeSlugs.map(function makePath(name) {
   return './' + name + '/**';
 });
-var srcFiles = recipes.map(function makeSourceFile(name) {
-  return './' + name + '/*.js';
-});
+
+var recipes = (function() {
+  var cache;
```
Comment (digitarald) [inline gulpfile.js]: is the lexing so slow that we need to cache?

File: gulpfile.js [hunk 3]
```diff
@@ -1,23 +1,120 @@
...
 });
-var srcRecipes = recipes.map(function makePath(name) {
+var srcRecipes = recipeSlugs.map(function makePath(name) {
   return './' + name + '/**';
 });
-var srcFiles = recipes.map(function makeSourceFile(name) {
-  return './' + name + '/*.js';
-});
+
+var recipes = (function() {
```
Comment (digitarald) [inline gulpfile.js]: could this be in a separate file?

File: gulpfile.js [hunk 4]
```diff
@@ -30,25 +127,69 @@ gulp.task('lint', function lint() {
...
+      return template.writeFile('./dist/' + recipe.slug + '.html', content, { currentRecipe: recipe });
+    });
+  }));
+});
+
+gulp.task('build:demos', ['clean'], function buildRecipes() {
+  return Promise.all(recipes().map(function(recipe) {
+    return renderFile('./src/tpl/demo.html', { recipe: recipe })
+    .then(function(output) {
+      return template.writeFile('./dist/' + recipe.slug + '_demo.html', output, { currentRecipe: recipe });
```
Comment (digitarald) [inline gulpfile.js]: Don't we loose the folder? We need it for the right sw scope.

Comment (brendandahl) [inline gulpfile.js]: They're iframe'd.

File: gulpfile.js [hunk 5]
```diff
@@ -1,23 +1,120 @@
...
+      return cache;
+    }
+    cache = [];
+    recipeSlugs.forEach(function(recipe) {
+      var tokens = marked.lexer(fs.readFileSync(recipe + '/README.md', 'utf8'));
+      if (!tokens.length) {
+        console.warn(recipe + ': is missing title');
+      }
+      var name = tokens[0].text;
+      if (!name.startsWith('Recipe: ')) {
```
Comment (digitarald) [inline gulpfile.js]: It seems kind of redundant to add it in each readme. What is the benefit of enforcing that?

File: gulpfile.js (hunk 3 above)
Comment (brendandahl) [inline gulpfile.js]: Yes, it's in my todo above. I started doing it, but it requires a change or two so I was going to wait.

**Comment by digitarald** [comment]
re sass: no strong feelings against any choice, especially as foundation encourages to be used with sass. We could throw in the same mix as platatus with cssnext (there is a big movement from sass to cssnext interestingly).

File: gulpfile.js (hunk 5 above)
Comment (brendandahl) [inline gulpfile.js]: I just needed it to be consistent so titles didn't look bad. I'm open to changing it to not having the 'Recipe' part.

**Comment by digitarald** [comment]
r+ :+1: just had nits, ignore for now

**Comment by darkwing** [comment]
I got the following when trying to build:

```
[16:23:06] Using gulpfile ~/Projects/serviceworker-cookbook/gulpfile.js
[16:23:06] Task 'buld' is not in your gulpfile
[16:23:06] Please check the documentation for proper gulpfile formatting
dwalsh-29224:serviceworker-cookbook dwalsh$ gulp build
[16:23:09] Using gulpfile ~/Projects/serviceworker-cookbook/gulpfile.js
[16:23:09] Starting 'clean'...
[16:23:09] Finished 'clean' after 6.28 ms
[16:23:09] Starting 'build:index'...
Recipe: api-analytics must have title and summary
Recipe: cache-then-network must have title and summary
Recipe: immediate-claim must have title and summary
Recipe: message-relay must have title and summary
Recipe: offline-fallback must have title and summary
Recipe: push-clients must have title and summary
Recipe: push-payload must have title and summary
Recipe: push-rich must have title and summary
Recipe: push-simple must have title and summary
Recipe: push-subscription-management must have title and summary
[16:23:09] Starting 'build:intros'...
[16:23:09] 'build:intros' errored after 455 μs
[16:23:09] TypeError: undefined is not a function
    at /Users/dwalsh/Projects/serviceworker-cookbook/gulpfile.js:38:17
    at Array.forEach (native)
    at /Users/dwalsh/Projects/serviceworker-cookbook/gulpfile.js:32:17
    at Gulp.buildRecipes (/Users/dwalsh/Projects/serviceworker-cookbook/gulpfile.js:175:22)
    at module.exports (/Users/dwalsh/Projects/serviceworker-cookbook/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/dwalsh/Projects/serviceworker-cookbook/node_modules/gulp/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/dwalsh/Projects/serviceworker-cookbook/node_modules/gulp/node_modules/orchestrator/index.js:214:10)
    at /Users/dwalsh/Projects/serviceworker-cookbook/node_modules/gulp/node_modules/orchestrator/index.js:279:18
    at finish (/Users/dwalsh/Projects/serviceworker-cookbook/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:21:8)
    at cb (/Users/dwalsh/Projects/serviceworker-cookbook/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:29:3)
[16:23:09] Finished 'build:index' after 20 ms
```

**Comment by darkwing** [comment]
I updated to 4.x and everything worked though, so should be an easy fix.

**Comment by brendandahl** [comment]
> at /Users/dwalsh/Projects/serviceworker-cookbook/gulpfile.js:38:17

Oh probably is the startsWith() call. I guess we could just require a certain node version for this project.

**Pull request merged**